presto-native-execution/src/test/java/com/facebook/presto/nativeworker/ContainerQueryRunnerUtils.java

```diff
@@ -0,0 +1,332 @@
+        createPropertiesFile("testcontainers/coordinator/etc/node.properties", properties);
+    }
+
+    public static void createNativeWorkerNodeProperties(String nodeId)
+    {
+        Properties properties = new Properties();
+        properties.setProperty("node.environment", "testing");
+        properties.setProperty("node.location", "testing-location");
+        properties.setProperty("node.id", nodeId);
+        createPropertiesFile("testcontainers/" + nodeId + "/etc/node.properties", properties);
+    }
+
+    public static void createNativeWorkerVeloxProperties(String nodeId)
+    {
+        Properties properties = new Properties();
+        properties.setProperty("mutable-config", "true");
+        createPropertiesFile("testcontainers/" + nodeId + "/etc/velox.properties", properties);
+    }
+
+    public static void createCoordinatorEntryPointScript()
+    {
+        String scriptContent = "#!/bin/sh\n" +
+                "set -e\n" +
+                "$PRESTO_HOME/bin/launcher run\n";
+        createScriptFile("testcontainers/coordinator/entrypoint.sh", scriptContent);
+    }
+
+    public static void createNativeWorkerEntryPointScript(String nodeId)
+    {
+        String scriptContent = "#!/bin/sh\n\n" +
+                "GLOG_logtostderr=1 presto_server \\\n" +
+                "    --etc-dir=/opt/presto-server/etc\n";
+        createScriptFile("testcontainers/" + nodeId + "/entrypoint.sh", scriptContent);
+    }
+
+    public static void deleteDirectory(String directoryPath)
+    {
+        File directory = new File(directoryPath);
+        deleteDirectory(directory);
+    }
+
+    private static void deleteDirectory(File directory)
+    {
+        if (directory.isDirectory()) {
+            File[] files = directory.listFiles();
+            if (files != null) {
+                for (File file : files) {
+                    deleteDirectory(file);
+                }
+            }
+        }
+        if (directory.delete()) {
+            logger.info("Deleted: " + directory.getPath());
+        }
+        else {
+            System.err.println("Failed to delete: " + directory.getPath());
+        }
+    }
+
+    public static void createPropertiesFile(String filePath, Properties properties)
+    {
+        try {
+            File file = new File(filePath);
+
+            File parentDir = file.getParentFile();
+            if (!parentDir.exists()) {
+                parentDir.mkdirs();
+            }
+
+            if (file.exists()) {
+                throw new IOException("File exists: " + filePath);
+            }
+
+            try (FileWriter writer = new FileWriter(file)) {
+                for (String key : properties.stringPropertyNames()) {
+                    writer.write(key + "=" + properties.getProperty(key) + "\n");
+                }
+                logger.info(file.getName() + " created successfully with content : " + properties);
+            }
+        }
+        catch (IOException io) {
```

Comment:
should this just bubble up and fail the test?

Reply:
updated the code